sw/device/lib/runtime/ibex.h

```diff
@@ -0,0 +1,40 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_RUNTIME_IBEX_H_
+#define SW_DEVICE_LIB_RUNTIME_IBEX_H_
+
+#include <stddef.h>
+
+#include "sw/device/lib/base/stdasm.h"
+
+/**
+ * This header provides Ibex-specific functions, such as the clock frequency and
+ * cycle-accurate busy loops.
+ */
+
+/**
+ * The clock frequency of the Ibex core, in hertz.
+ */
+extern const size_t kIbexClockFreqHz;
+
+/**
+ * Spins for roughly eight cycles per loop.
+ *
+ * This function should not be used for time-keeping.
+ *
+ * @arg loops the number of cycles to burn, divided by eight.
```

Comment:
I would much prefer having `cycles` be the argument and documenting "must be a multiple of eight", rather than requiring the caller to do the division. I realise it means the `ibex_busy_loop` will now take more cycles, but I think we're calling it with large enough numbers that busy looping for a few cycles more is not an issue (indeed the implementation could avoid the division overhead by using `addi %1, %1, -8; blez %1, busy_loop%=`).

Reply:
Good point. Done.